### PR TITLE
[Backport][ipa-4-6] ipatests: add test_trust suite to nightly runs

### PR DESCRIPTION
**Pull request by wladich**
The test suite test_trust was missing in nightly definitions
because PR-CI was not able to provision multi-AD topology.
Now that PR-CI is updated, we can start executing this test suite.
It is not reasonable to add it to gating as this suite is
time consuming like other tests requiring provisioning of AD instances.

This PR brings backports and fixes for the test_trust suite:
* use TLS connection for LDAP
* provide AD admin password when trying to establish trust with unreachable AD DC
* adapt expected error message from `ipa trust-add` when trying to establish trust with unreachable AD DC
* Add workaround for known bug in SSSD

### CODE REVIEW

**Comment by flo-renaud** [comment]
Hi @wladich 
the test failed because of a missing command (`ntpdate`). On ipa-4-7+ branches the `tasks.sync_time()` method is based on chronyd but on ipa-4-6 branch it relies on ntpdate.

**Comment by wladich** [comment]
I will add a backport for switching from ntpdate to chronyd since Fedora 27+ and RHEL 7.6+ do use chrony.

**Comment by wladich** [comment]
New strange failures found. Postponing the PR

File: ipatests/test_integration/test_trust.py [hunk 1]
```diff
@@ -635,7 +635,8 @@ def test_server_option_with_unreachable_ad(self):
             # This checks that our setup is correct
             result = self.master.run_command(
                 ['ipa', 'trust-add', self.ad.domain.name,
-                 '--admin', 'Administrator', '--password'], raiseonerr=False)
+                 '--admin', 'Administrator', '--password'], raiseonerr=False,
+                stdin_text=self.master.config.ad_admin_password)
```
Comment (flo-renaud) [inline ipatests/test_integration/test_trust.py]: Is this required only on ipa-4-6 branch? I could not spot the code that would explain this difference.

File: ipatests/test_integration/test_trust.py [hunk 2]
```diff
@@ -501,8 +501,24 @@ def test_user_gid_uid_resolution_in_forest_trust_with_shared_secret(self):
...
+            result = self.master.run_command(['getent', 'passwd', testuser],
+                                             ok_returncode=[0, 2])
+            try:
+                assert result.returncode == 0
+                assert re.search(
+                    testuser_regex, result.stdout_text), result.stdout_text
+                break
+            except AssertionError:
+                if time.time() - start > timeout:
+                    raise
```
Comment (flo-renaud) [inline ipatests/test_integration/test_trust.py]: You can add a sleep(1) if we didn't reach the timeout yet.

Comment (wladich) [reply ipatests/test_integration/test_trust.py]: oops... missed the most important part

File: ipatests/test_integration/test_trust.py (hunk 1 above)
Comment (wladich) [reply ipatests/test_integration/test_trust.py]: 1. I admit this is a mistake in tests.
2. In master this mistake does not prevent test from passing: http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/7aafc9e4-68be-11ea-97e2-fa163e3ad6c5/report.html
```
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd544:transport.py:513 RUN ['ipa', 'trust-add', 'ad.test', '--admin', 'Administrator', '--password']
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd544:transport.py:558 ipa: ERROR: CIFS server communication error: code "3221225653", message "{Device Timeout} The specified I/O operation on %hs was not completed before the time-out period expired." (both may be "None")
```

In master branch empty password does produce error message when AD controller is available. So it looks as if the checks for password and connectivity are made in different order. Unfortunately I do not know if it is a regression or intended change (and not quite sure if that is the real root cause).

I will add a separate PR for master, ipa-4-7 and ipa-4-8 for the fix but I am not sure how to deal with this behavior change.

Comment (wladich) [reply ipatests/test_integration/test_trust.py]: Created #4399

**Comment by wladich** [comment]
Problem installing packages on master and client: http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/1ad60d62-6aa3-11ea-9e23-fa163ebb6b2b
Retrying.

**Comment by flo-renaud** [comment]
Hi @wladich 
LGTM. Please remove the temp commit and I will ack when prci completes.

**Comment by wladich** [comment]
@flo-renaud 
All tests are green

**Comment by flo-renaud** [comment]
Hi @wladich 
thanks for the heads-up, ACK.

**Comment by wladich** [comment]
ipa-4-6:

* 2b6638becbfbae746cef35176890ae3f4a8b01a6 TestBasicADTrust.test_ipauser_authentication
* 795a973c00c2fe862b1eff8bd851d8eafe9d970a ipatests: provide AD admin password when trying to establish trust
* 941c231b692216f3dc4b66944dd170b5380fe981 ipatests: use less strict check for error message
* 37e383aae94b0450c06f0e78354245e4b14d70f5 ipatests: add workaround for unfixed sssd bug in Fedora 27
* d44374e761a1e7f5aaca22399631f77fccc45f94 ipatests: add test_trust suite to nightly runs

**Pull request closed without merging**